lib/uklibparam/include/uk/libparam.h

```diff
@@ -187,18 +130,16 @@ UK_CTOR_PRIO(UK_LIBPARAM_LIBDESC_CTOR, 0);
 
 #define __UK_LIBPARAM_PARAM_DEFINE(arg_var, arg_addr, arg_type, arg_count, \
 				   arg_desc)				\
-	static struct uk_libparam_param __UK_LIBPARAM_PARAM_NAME(arg_var) = { \
-		.name  = STRINGIFY(arg_var),				\
-		.desc  = arg_desc,					\
-		.type  = _UK_LIBPARAM_PARAM_TYPE_##arg_type,		\
-		.count = arg_count,					\
-		.addr  = arg_addr					\
+	static const struct uk_libparam_param				\
+	__used __section("." STRINGIFY(UK_LIBPARAM_PARAMSECTION_NAME))	\
+	__align(8) __UK_LIBPARAM_PARAM_NAME(arg_var) = {		\
+		.prefix = STRINGIFY(UK_LIBPARAM_LIBPREFIX),		\
```

Comment:
Out of curiosity, is `gcc` clever enough to declare the library prefix only once and use the same reference for each parameter of the same library? I think it is, because of `const char *` and the actual string ending up in `rodata`, correct?

Reply:
Exactly, when adding a 2nd parameter in `libuknetdev`, its `prefix` element will point the same string in `.rodata` (`0x0015d12d` at `0x0016f650` for the 1st parameter, and `0x0015d12d` at `0x0016f688` for the 2nd parameter):
```
Hex dump of section '.uk_libparam_params':
  0x0016f650 2dd11500 00000000 6cda1500 00000000 -.......l.......
  0x0016f660 08dd1500 00000000 0b000000 00000000 ................
  0x0016f670 04000000 00000000 a0c41a00 00000000 ................
  0x0016f680 00000000 00000000 2dd11500 00000000 ........-.......
  0x0016f690 b0eb1500 00000000 08dd1500 00000000 ................
  0x0016f6a0 0b000000 00000000 04000000 00000000 ................
  0x0016f6b0 c0c41a00 00000000 00000000 00000000 ................
  0x0016f6c0 82d11500 00000000 f3bb1500 00000000 ................
  0x0016f6d0 a6101600 00000000 0b000000 00000000 ................
  0x0016f6e0 04000000 00000000 a0fb1a00 00000000 ................
  0x0016f6f0 00000000 00000000                   ........
```
In fact, GCC is so smart that for the `netdev` prefix specified in `addlibparamprefix`, it will reuse part of the existing `libuknetdev` string in `rodata` at `0x0015d128`:
```
  0x0015d120 6962756b 6d706900 6c696275 6b6e6574 ibukmpi.libuknet
  0x0015d130 64657600 6c696275 6b72616e 646f6d00 dev.libukrandom.
```

Reply:
Thanks for confirming 👍🏻